src/TRestRawSignal.cxx

```diff
@@ -591,12 +595,14 @@ void TRestRawSignal::GetWhiteNoiseSignal(TRestRawSignal* noiseSignal, Double_t n
     double* dd = new double();
     uintptr_t seed = (uintptr_t)dd + (uintptr_t)this;
     delete dd;
-    TRandom3* fRandom = new TRandom3(seed);
+    TRandom3 random(seed);
```

Comment:
Perhaps the `seed` should be stored inside `TRestRawSignal::fSeed`. It should be initialised just once, not every time `GetWhiteNoiseSignal` is called. This may lead to unreproducible results.

Reply:
Implemented in [9499c8e](https://github.com/rest-for-physics/rawlib/pull/114/commits/9499c8e6729e22f4034240cbf41e8d58f1b89666).

By default it will use the same seed as the global seed for `TRandom` but it can also be specified via `TRestRawSignal::SetSeed`.